pixelator.html

```diff
@@ -1,126 +1,172 @@
-        const pixelvalues = imageData.data;
-        // console.log(pixelvalues)
-
-        ctx.clearRect(0, 0, canvas.width, canvas.height);
-
-        // lööp over image, each 5th pixel
-        let loopkiller = 100000; // around 100k
-
-        for (let row = 1; row < imageHeight; row = row + pixelMultiplier) {
-          for (let col = 1; col < imageWidth; col = col + pixelMultiplier) {
-            loopkiller--;
-            if (loopkiller < 0) { return console.log("killed"); }
-            
-            // get 1st pixel color
-            const pixelReference = ((row*imageWidth)+(col))*4;
-            if (pixelReference === null || typeof pixelReference === "undefined") {return console.log("Invalid pixel reference"); }
-
-            const color = `rgb(
+<head>
+  <meta charset="UTF-8">
+  <meta http-equiv="X-UA-Compatible" content="IE=edge">
+  <meta name="viewport" content="width=device-width, initial-scale=1.0">
+  <meta name="author" content="Joel Peltonen">
+  <title>Document</title>
+  <style>
+    * {
+      margin: 0;
+      padding: 0;
+    }
+
+    body {
+      display: flex;
+      flex-direction: column;
+      justify-content: center;
+      align-items: center;
+      width: 100vw;
+      height: 100vh;
+    }
+
+    #controls {
+      display: flex;
+      flex-direction: column;
+      justify-self: center;
+      text-align: right;
+      background-color: whitesmoke;
+    }
+
+    .input-group {
+      display: flex;
+      align-items: center;
+      margin: .5em;
+    }
+
+    label {
+      width: 10em;
+      margin-right: .5em;
+    }
+
+    #graphics {
+      display: flex;
+      justify-content: center;
+      align-items: center;
+    }
+  </style>
+</head>
+
+<body onload="initPage()">
+<div id="controls">
+  <div class="input-group">
+    <label for="imageFile">Image File:</label>
+    <input type="file" name="" id="imageFile">
+  </div>
+  <div class="input-group">
+    <label for="multiplierInput">Pixelation Factor:</label>
+    <input type="number" id="multiplierInput" min="0" max="666" value="2" />
+  </div>
+</div>
+<div id="graphics">
+  <img id="sourceImage" src="" alt="source image"/>
+  <canvas id="targetCanvas"></canvas>
+</div>
+Joel Peltonen 2021
+
+<script type="text/javascript">
+
+  const initialFilename = 'Pink-Gonzo.jpg';
+  const maxFileSize = 900000;
+  const maxPixelationFactor = 666;
+  const minPixelationFactor = 2;
+  const initialPixelationFactor = 5;
+  let pixelationFactor = initialPixelationFactor;
+
+  // Are you KIDDING me with this type casting?
+  const canvas = /** @type {HTMLCanvasElement} */ document.querySelector('#targetCanvas');
+  const ctx = /** @type {CanvasRenderingContext2d} */ canvas.getContext('2d');
+  const fileDialog = document.querySelector("#imageFile");
+  const sourceImage = document.querySelector('#sourceImage');
+  const multiplierInput = document.querySelector("#multiplierInput");
+
+  // init page (triggered by body onload event)
+  function initPage() {
```

Comment:
Yeah, think I understand. I probably more prefer to have separate functions even if they blow up code a bit. Two thoughts about that.
a) If I use an arrow function within another function, I cannot call the function from somewhere else because it has no name and is out of scope. In this project the fileRead should be triggered from two locations (page load / file change), I thought. Here we probably have no need to call initPage twice.
b) I often like SoC. The (arrow) onload-handler you defined in your fileread-section has more to do with the image than the read functionality, I thought. So I dragged it out:-)